modules/transactions/src/main/java/org/apache/ignite/internal/tx/impl/TxManagerImpl.java

```diff
@@ -666,7 +673,7 @@ private CompletableFuture<Void> makeFinishRequest(
                             new TxStateMeta(
                                     txResult.transactionState(),
                                     localNodeId,
-                                    old.commitPartitionId(),
+                                    old == null ? null : old.commitPartitionId(),
```

Comment:
seems you need change your interface :
<T extends TxStateMeta> T updateTxMeta(UUID txId, Function<@Nullable TxStateMeta, TxStateMeta> updater);

Reply:
![image](https://github.com/apache/ignite-3/assets/1107510/031272f2-bc80-45ba-a05a-7df45b7f23a2)
and voila )